### ISSUE TITLE
Custom Mixins

### ISSUE THREAD

**Issue by drewwells** [sass/libsass#960]
libsass has the ability to specify custom functions, but not custom mixins (that I'm aware of). Are there plans to enable specifying custom mixins?

I'm hacking custom mixins by either injecting mixins at the beginning of source string or using the custom importer interface. The second one is fine, but does require the developer to include special `@import` to enable the custom mixin.

Here's an example mixin that gets injected into the source:

```
@mixin sprite-dimensions($map, $name) {
  $file: sprite-file($map, $name);
  height: image-height($file);
  width: image-width($file);
}
```

**Comment by mgreter** [comment]
This basically boiled down to have custom headers. I implemented something experimental in #1000 to support custom headers. I will paste the relevant parts from the linked issue:

Custom headers (#960) act like importers, but only run once at the beginning of each compilation. They are primarily meant to define custom mixins (via plugins or context options). Beside that they might be usefull for other cool things I currently cannot think of :smile: Opposite to custom importers, all custom headers will be executed in priority order and all imports will be accumulated (so many custom headers can add various custom mixins or css-code).

**Comment by drewwells** [comment]
That is AWESOME! Exactly what I was looking for!

**Comment by mgreter** [comment]
Merged #1000 which contains a first experimental implementation!
